packages/amplify-category-notifications/src/multi-env-manager.ts

```diff
@@ -370,3 +499,54 @@ const fillTeamProviderInfo = (context: $TSContext, migrationInfo: $TSAny) => {
     Object.assign(teamProviderInfo[migrationInfo.envName].categories, categoryTeamInfo);
   }
 };
+
+const buildPinpointInputParametersFromAmplifyMeta = (context: $TSContext): Record<string, $TSAny> => {
+  const { envName } = context.exeInfo.localEnvInfo;
+  const { amplifyMeta } = context.exeInfo;
+  const pinpointInputParameters: Record<string, $TSAny> = { envName };
+  const categoryMeta = amplifyMeta[AmplifyCategories.NOTIFICATIONS];
+  const availableChannels = getAvailableChannels();
+  if (!categoryMeta) {
+    return pinpointInputParameters;
+  }
+
+  const resourceNames = Object.keys(categoryMeta);
+  for (const resourceName of resourceNames) {
+    const resource = categoryMeta[resourceName];
+    if (resource.service === AmplifySupportedService.PINPOINT) {
+      pinpointInputParameters.service = AmplifySupportedService.PINPOINT;
+      if (resource.output) {
+        for (const channelName of availableChannels) {
+          // eslint-disable-next-line max-depth
```

Comment:
should we increase max-depth setting or break it down ?

Reply:
re-wrote to fit into the max-depth setting